File: src/main/java/ma/enset/accountscqrseventsourcingaxon/commonapi/events/AccountCreatedEvent.java
package ma.enset.accountscqrseventsourcingaxon.commonapi.events;

import lombok.Getter;
import ma.enset.accountscqrseventsourcingaxon.commonapi.enums.AccountStatus;

public class AccountCreatedEvent extends BaseEvent<String>{
    @Getter private double balance;
    @Getter private String currency;
    @Getter private AccountStatus status;
    public AccountCreatedEvent(String id, double balance, String currency, AccountStatus status) {
        super(id);
        this.balance = balance;
        this.currency = currency;
        this.status = status;
    }
}

File: src/main/java/ma/enset/accountscqrseventsourcingaxon/commonapi/events/AccountCreatedEvent.kt
package ma.enset.accountscqrseventsourcingaxon.commonapi.events

import ma.enset.accountscqrseventsourcingaxon.commonapi.enums.AccountStatus

class AccountCreatedEvent(id: String, balance: Double, currency: String, status: AccountStatus) : BaseEvent<String>(id) {
    val balance: Double = balance
    val currency: String = currency
    val status: AccountStatus = status
}
